### PR TITLE
Install system version-aware and PolyChord update

### LINKED ISSUE #197: No way right now to detect installs of old data
The new BAO needs an update of the `data` repo, but `cobaya-install` right now fails to detect that this update is needed when installing on top of already installed data (so `--force` is required).

The first step to solve this would be to store the currently installed version somewhere:
- either in the `data` repo in a way that is not removed when downloaded (e.g. using a version badge via https://naereen.github.io/badges/)
- or create a file called sth like `version.dat` at download time by our `download_github_release` function.

I lean towards the 2nd one. @cmbant?

### PR DESCRIPTION
**Pull request by JesusTorrado**
Making the `install` function version-aware.

- [x] GitHub downloads write a `version.dat` file containing version tag.
- [x] `install()` detects obsolete versions and lists obsolete packages at the end (if no version detected: obsolete)
- [x] `install()` gets `(--)upgrade` option. Off by default: don't overwrite possible user-changes.
- [x] non-github: Planck clik
- [x] github assets: Planck native: lite/CamSpec...
- [x] non-github: Bicep (nothing to do)
- [x] early install and version checks at run time for data
- [x] test camspec and unbinned
- [x] CAMB
- [x] CLASS
- [x] PolyChord
  - [x] merge #232
  - [x] wait for new minor version? (see #232) -- if so, restore version check
  - [x] tests

Closes #197

### CODE REVIEW

**Comment by JesusTorrado** [comment]
@cmbant What's the purpose of the following line?
https://github.com/CobayaSampler/cobaya/blob/8152a094cc23da1506d6848fb67327a8b2740210/cobaya/tools.py#L234
It is run under a `if allow_external and not component_path`, and if it does not raise an exception, `return_class` will be run below, which calls `load_module`, which in turns does the exact same `import_module` call, without any error handling in between, apparently. So why would't the current code,

    try:
        import_module(module_name)
    except Exception:
        exc_info = sys.exc_info()
    else:
        try:
            return return_class(module_name)
        except:
            exc_info = sys.exc_info()

be equivalent to the simpler following one?

    try:
        return return_class(module_name)
    except:
        exc_info = sys.exc_info()

(probably I am wrong, but would like to understand why)

**Comment by cmbant** [comment]
I don't recall exactly, but  probably to give clearer error message and shorter stack trace. Note `exc_info[0] is ModuleNotFoundError ` line further down.

**Comment by JesusTorrado** [comment]
I just checked that the code is indeed equivalent if it would raise an exception at import time, except a possible reload, which is not relevant here. I have substituted below inside the `get_class` code the lines that would actually run up to `import_module` in both cases:

    try:
        import_module(module_name)
    except Exception:
        exc_info = sys.exc_info()
    else:
        try:
            if name in sys.modules and reload:
                del sys.modules[name]
            return import_module(name)
        except:
            exc_info = sys.exc_info()

I am going to simplify this one in principle, and do some testing

File: cobaya/component.py [hunk 1]
```diff
@@ -223,3 +464,184 @@ def __enter__(self):
...
+    tried = " and ".join(allow_internal * ["internal"] + allow_external * ["external"])
+    add_msg = f" Tried loading {tried} classes. No component path was given."
+    raise ComponentNotFoundError(_not_found_msg + add_msg)
+
+
+def module_class_for_name(m, name):
+    """Get Camel- or uppercase class name matching name in module m."""
+    result = None
+    valid_names = {name, name[:1] + name[1:].replace('_', '')}
+    from cobaya.component import CobayaComponent
```
Comment (cmbant) [inline cobaya/component.py:629]: Importing itself

File: cobaya/theories/camb/camb.py [hunk 2]
```diff
@@ -220,8 +220,15 @@ def initialize(self):
         allow_global = not self.path
         if not self.path and self.packages_path:
             self.path = self.get_path(self.packages_path)
-        self.camb = self.is_installed(path=self.path, allow_global=allow_global,
-                                      check=False)
+        min_version = None if self.ignore_obsolete else self._min_camb_version
```
Comment (cmbant) [inline cobaya/theories/camb/camb.py:223]: ignore_obsolete appears not defined or declared

File: cobaya/output.py [hunk 3]
```diff
@@ -359,7 +359,7 @@ def check_and_dump_info(self, input_info, updated_info, check_compatible=True,
                             updated_info[k][c]["version"] = old_version
                             updated_info_trimmed[k][c]["version"] = old_version
                         elif old_version is not None:
-                            cls = get_resolved_class(
+                            cls = get_component_class(
                                 c, k, None_if_not_found=True,
```
Comment (cmbant) [inline cobaya/output.py:363]: Unknown argument

**Comment by JesusTorrado** [comment]
Also, I think there was no error

**Comment by cmbant** [comment]
Upgrading old versions I think can fail because this raises an error (old installs don't have .dat files):
```

            elif opts.get("github_release"):
                try:
                    with open(os.path.join(path, _version_filename), "r") as f:
                        installed_version = version.parse(f.readlines()[0])
                except FileNotFoundError:  # old install: no version file
                    raise VersionCheckError("Could not read current version.")
```

Trace

`(base) [aml1005@login-e-13 cobaya]$ cobaya-install --upgrade planck_2018_lowl.EE
[install] Installing external packages at '/home/aml1005/git/cobaya/modules'
[install] The installation path has been written into the global config file: /home/aml1005/.config/cobaya/config.yaml

================================================================================
planck_2018_lowl.EE
================================================================================

[install] Checking if dependencies have already been installed...
[install] Version check failed: Could not read current version.
[install] Installing...
[clik] `clik` module loaded successfully from /rds/user/aml1005/hpc-work/modules/code/planck/code/plc_3.0/plc-3.01/lib/python/site-packages/clik
Traceback (most recent call last):
  File "/home/aml1005/git/cobaya/cobaya/likelihoods/base_classes/InstallableLikelihood.py", line 98, in is_installed
    with open(os.path.join(path, _version_filename), "r") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/rds/user/aml1005/hpc-work/modules/data/planck_supp_data_and_covmats/version.dat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/aml1005/git/cobaya/cobaya/install.py", line 215, in install
    success = install_this(path=general_abspath, **kwargs_install)
  File "/home/aml1005/git/cobaya/cobaya/likelihoods/base_classes/planck_clik.py", line 207, in install
    if force or not cls.is_installed(path=path, code=False, data=True):
  File "/home/aml1005/git/cobaya/cobaya/likelihoods/base_classes/planck_clik.py", line 171, in is_installed
    result &= native.is_installed(**kwargs)
  File "/home/aml1005/git/cobaya/cobaya/likelihoods/base_classes/InstallableLikelihood.py", line 101, in is_installed
    raise VersionCheckError("Could not read current version.")
cobaya.tools.VersionCheckError: Could not read current version.
[install] *ERROR* An unknown error occurred. Delete the external packages folder '/home/aml1005/git/cobaya/modules' and try again. Please, notify the developers if this error persists.
[install] *ERROR* Installation failed! Look at the error messages above. Solve them and try again, or, if you are unable to solve them, install the packages required by this component manually.

================================================================================
* Summary *
================================================================================

[install] *ERROR* The installation (or installation test) of some component(s) has failed:
 - likelihood:planck_2018_lowl.EE
Check output of the installer of each component above for precise error info.

[exception handler] ---------------------------------------

Traceback (most recent call last):
  File "/rds/user/aml1005/hpc-work/anaconda2/bin/cobaya-install", line 11, in <module>
    load_entry_point('cobaya', 'console_scripts', 'cobaya-install')()
  File "/home/aml1005/git/cobaya/cobaya/install.py", line 549, in install_script
    "no_set_global", "skip", "skip_global", "debug", "upgrade"]})
  File "/home/aml1005/git/cobaya/cobaya/install.py", line 278, in install
    bullet + bullet.join(failed_components))
cobaya.log.LoggedError: The installation (or installation test) of some component(s) has failed:
 - likelihood:planck_2018_lowl.EE
Check output of the installer of each component above for precise error info.

-------------------------------------------------------------

`

**Pull request merged**